cpp/include/cudf/detail/iterator.cuh

```diff
@@ -89,6 +91,10 @@ struct value_accessor
  *
  * Dereferencing the returned iterator for element `i` will return `column[i]`
  * if it is valid, or `null_replacement` if it is null.
+ * This iterator is only allowed for nullable columns.
+ *
+ * @throws `cudf::logic_error` if the column is not nullable.
+ * @throws `cudf::logic_error` if column datatype and Element type mismatch.
```

Comment:
This is a good idea, but I don't see where it is enforced.

Reply:
'column is not nullable', is enforced inside constructor `value_accessor`.
`column datatype and Element type mismatch` is enforced in `make_null_replacement_iteraor`

Reply:
Ah, I missed the `Element` check. Thanks.